### PR TITLE
Added geometry.py

### PR DESCRIPTION
**Pull request by geomatician**
Before you submit a pull request, please fill in the following:

**Related Issue(s):**
https://github.com/stac-utils/stactools/issues/184

**Description:**
Added geometry.py to Core to generate a bounding box from a GeoJSON geometry, STAC Item, or FeatureCollection

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

**Comment by gadomski** [comment]
@pjhartzell can you take a look and see if there's any other geometry helpers that you think could fit into this PR?

**Comment by gadomski** [comment]
> Off-topic: Note that reproject_geom suffers from the problem that reprojecting a geometry generally reduces its fidelity (between vertices) to the originally described shape. Closely related to one of the challenges in the upcoming footprint creator.

@pjhartzell worth opening its own issue for?

**Comment by pjhartzell** [comment]
@gadomski I can open one up.

File: src/stactools/core/geometry.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from typing import Any, Dict, List
+
+import rasterio.features
+
+
+def bounding_box(geom: Dict[str, Any]) -> List[float]:
+    """Extracts and returns the bounding box of a GeoJSON geometry.
+
+    Args:
+        geom (dict): A STAC Item, FeatureCollection, or GeoJSON geometry
```
Comment (philvarner) [inline src/stactools/core/geometry.py:10]:
```suggestion
        geom (dict): A GeoJSON Feature, GeoJSON FeatureCollection, GeoJSON geometry, STAC Item, or STAC ItemCollection.
```

**Review by philvarner — APPROVED** [review]

**Pull request merged**